android/engine/src/main/java/org/smartregister/fhircore/engine/rulesengine/RulesFactory.kt

```diff
@@ -417,6 +387,9 @@ constructor(
         ?.find { parentResourceId.equals(it.parentResourceId, ignoreCase = true) }
         ?.count
         ?: 0
+
+    fun getDateFilterFromCriteria(criteria: String) =
```

Comment:
This function can be renamed to something more meaningful. It's only used to minus years, can we make it work with addition too? The date format should be provided as an argument to make it more flexible (default format `SDF_YYYY_MM_DD`)